### PR TITLE
support other CRAN mirrors

### PR DESCRIPTION
**Pull request by katrinleinweber**
Hi!

Since [CRAN.RStudio.com](https://cran.rstudio.com/) is a very popular mirror, please consider expanding the translator to it. Related to #1487.

Also, would a minimal reg-ex to cover even other mirrors be advisable? `cran\..+/web/packages` might work.

Cheers :-)

### CODE REVIEW

**Comment by adam3smith** [comment]
Generally happy with this & extending it to other mirrors. Note you'll have to bump the timestamp for updates to propagate (Scaffold, which we recommend, would do that automatically)

File: R-Packages.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"translatorID": "24a10ebf-ada1-4b8d-8f76-5a29e24d3e78",
 	"label": "R-Packages",
 	"creator": "Sebastian Karcher",
-	"target": "r-project\\.org/web/packages/(.+/index\\.html|available_packages_by_(name|date)\\.html)",
+	"target": "(r-project\\.org|rstudio\\.com)/web/packages/(.+(/index\\.html)?|available_packages_by_(name|date)\\.html)",
```
Comment (adam3smith) [inline R-Packages.js]: once you make index.html optional, you can just delete the entire regex after packages: `.+(/index\\.html)?` literally matches anything.

**Comment by adam3smith** [comment]
What's the form of the other mirrors? I thought we already covered those?

**Comment by katrinleinweber** [comment]
Oh! That was an unwarranted assumption it seems. I was thinking of https://www.r-pkg.org/pkg/ggplot2 but they seem to have an entirely different site structure.

**Comment by katrinleinweber** [comment]
I used https://duckduckgo.com/?q=cran+web+packages and https://www.google.de/search?q=cran+web+packages to find other popular mirrors.

https://mran.microsoft.com/ & https://www.r-pkg.org/ are out-of-scope for me currently.

**Comment by katrinleinweber** [comment]
I feel bad duplicating the test code so much. Does Scaffold support extracting multiple URLs and comparing each result against the same `items` set? Or against each other?

**Comment by adam3smith** [comment]
Don't worry about it -- we only very rarely test different sites with the same expected result, so not worth implementing dedicated functionality for that scenario, and the size of these tests is just completely negligible (and it's not like it's code that actually needs to be updated/maintained by humans)

**Comment by katrinleinweber** [comment]
I delivered it with my bare hands, but it's good to know I don't have to take care of it.  🤣

**Comment by katrinleinweber** [comment]
related to https://github.com/force11/force11-sciwg/issues/9

**Comment by zuphilip** [comment]
I rewrote it w/o the FW and update some things. Please have a look. Should be fine for merge.

**Comment by katrinleinweber** [comment]
Manual test imports look fine for me (Zotero 5.0.35.1 & Firefox 52.6.0). Thank you :-)

File: R-Packages.js [hunk 2]
```diff
@@ -125,26 +170,255 @@ var testCases = [
...
+						"lastName": "Firth",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "Steve",
+						"lastName": "Taylor",
+						"creatorType": "author"
+					},
+					{
+						"firstName": "R. Core",
```
Comment (adam3smith) [inline R-Packages.js]: we should recognize the R Core Team and turn it into an institutional author. Presumably that'll show up on a fair number of packages, so should be worth it.

Comment (zuphilip) [reply R-Packages.js]: Done in the new version.

**Comment by katrinleinweber** [comment]
Thank you :-)

**Comment by zuphilip** [comment]
Okay, it is merged now. Thank you all for the contributions here!

**Comment by zuphilip** [comment]
@katrinleinweber Can we close the corresponding issue or is there something still open? (I haven't followed the conversion there closely.)

**Comment by katrinleinweber** [comment]
#1487 is "related", but not completely corresponding to this ;-)

**Pull request merged**